### ISSUE TITLE
Assert error on OSX

### ISSUE THREAD

**Issue by vsimon** [SFTtech/openage#588]
```
...
INFO [T759559866708810737] Initialized SDL video subsystems.
Assertion failed: (current_context), function extract_native_context, file /Users/vicken/src/openage/libopenage/gui/guisys/private/platforms/context_extraction_cocoa.mm, line 18.
Abort trap: 6
```

**Comment by TheJJ** [comment]
Do we really have to do the context extraction in objective c?

**Comment by ChipmunkV** [comment]
Can just cast the `SDL_GLContext`, but it's not documented.

**Comment by TheJJ** [comment]
This should also be fixed by #828.
